cmd/state-handler/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+
+	clientset "github.com/nmstate/kubernetes-nmstate/pkg/client/clientset/versioned"
+	informers "github.com/nmstate/kubernetes-nmstate/pkg/client/informers/externalversions"
+	"github.com/nmstate/kubernetes-nmstate/pkg/nmstatectl"
+	"github.com/nmstate/kubernetes-nmstate/pkg/signals"
+	"github.com/nmstate/kubernetes-nmstate/pkg/utils"
+)
+
+var (
+	executionType = flag.String("execution-type", "", "\"controller|client\" Whether controller actively handling state changes OR only one-shot client should be started.")
+	kubeconfig    = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	master        = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	namespace     = flag.String("n", "", "The namespace where the CRDs are created. If left blank and running via pod, it will be taken from there.")
+	hostname      = flag.String("host", "", "Name of the host on which to enforce and report state. If left blank and running via pod, it will be taken from there.")
+)
+
+func main() {
+	flag.Parse()
+
+	switch *executionType {
+	case "":
+		panic("execution-type must be specified")
+	case "controller":
+		controller()
+	case "client":
+		client()
+	}
+}
+
+func controller() {
+	// set up signals so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler()
+
+	cfg, err := clientcmd.BuildConfigFromFlags(*master, *kubeconfig)
+	if err != nil {
+		klog.Fatalf("Error building kubeconfig: %s", err.Error())
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		klog.Fatalf("Error building kubernetes clientset: %s", err.Error())
+	}
+
+	nmstateClient, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		klog.Fatalf("Error building nmstate clientset: %s", err.Error())
+	}
```

Comment:
minor: BuildConfigFromFlags(), clientset.NewForConfig(), GetNsmespace() and GetHostName() are common to both client and controller, could be taken out of the specific functions.